File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/meeting/MeetingRequest.java
package com.portSrilanka.board_admin_backend.dto.meeting;

import com.portSrilanka.board_admin_backend.enums.MeetingType;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class MeetingRequest {
    private String title;
    private MeetingType type;
    private LocalDateTime meetingDateTime;
    private LocalDateTime targetDateTime;
    private String location;
    private String description;
    private Long categoryId;
    private Long subcategoryId;
    private Long createdByUserId;
}

File: board-admin-frontend/frontend/lib/features/meetings/model/meeting_request.dart
class MeetingRequest {
  final String title;
  final String type;
  final String meetingDateTime;
  final String? targetDateTime;
  final String? location;
  final String? description;
  final int categoryId;
  final int subcategoryId;
  final int createdByUserId;

  MeetingRequest({
    required this.title,
    required this.type,
    required this.meetingDateTime,
    this.targetDateTime,
    this.location,
    this.description,
    required this.categoryId,
    required this.subcategoryId,
    required this.createdByUserId,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'type': type,
      'meetingDateTime': meetingDateTime,
      'targetDateTime': targetDateTime,
      'location': location,
      'description': description,
      'categoryId': categoryId,
      'subcategoryId': subcategoryId,
      'createdByUserId': createdByUserId,
    };
  }
}
